File: 119-pascals-triangle-ii/pascals-triangle-ii.py
class Solution(object):
    def getRow(self, rowIndex):
        """
        :type rowIndex: int
        :rtype: List[int]
        """
        res=[[1]]
        for i in range(2,rowIndex+2):
            val=[1]*i
            for j in range(1,i-1):
                val[j]=res[-1][j-1]+res[-1][j]
            res.append(val)

        return res[-1]

File: 119-pascals-triangle-ii/pascals-triangle-ii.cpp
class Solution {
public:
    vector<int> getRow(int rowIndex) {
        vector<vector<int>> pascal={{1}};
        for(int i=2;i<rowIndex+2;++i){
            vector<int> row(i,1);
            for(int j=1;j<i-1;++j){
                row[j]=pascal.back()[j]+pascal.back()[j-1];
            }
            pascal.push_back(row);
        }
        return pascal[rowIndex];
    }
};